app/HutchManager/Data/Entities/DataSource.cs

```diff
@@ -4,4 +4,6 @@ public class DataSource
 {
   public string Id { get; set; } = string.Empty;
   public DateTimeOffset LastCheckin { get; set; }
+
+  public List<Agent> Agents { get; set; } = null!;
```

Comment:
this is a list so you can initialise it as `new()` instead of `null!`? Assuming EF is still happy with that, I think it's preferable for the compiler.